src/components/dashboard/jumbotron/CountdownTimer.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
+import './countdown.css';
+
+/**
+ * Displays Time untill upcoming trip
+ */
+
+export default class CountdownTimer extends Component {
+  /**************************** Helper functions ****************************/
+  getTimeLeft(start) {
+    let curr = new Date();
+    let timeDiff = start.getTime() - curr.getTime();
+    let daysleft = Math.floor(timeDiff / (1000 * 3600 * 24));
+    let hoursleft = Math.floor(timeDiff / (1000 * 3600) - daysleft * 24);
+    let minutesleft = Math.floor(
+      timeDiff / (1000 * 60) - daysleft * 24 * 60 - hoursleft * 60
+    );
+    let secondsleft = Math.floor(
+      timeDiff / 1000 -
+        daysleft * 24 * 60 * 60 -
+        hoursleft * 60 * 60 -
+        minutesleft * 60
+    );
+    return {
+      days: daysleft,
+      hours: hoursleft,
+      minutes: minutesleft,
+      seconds: secondsleft
+    };
+  }
+
+  /**
+   * Creates all HTML for countdown timer
+   */
+  getTimeDiv() {
+    let currTrip = this.props.currTripInfo.trip;
+    if (currTrip === null) {
```

Comment:
Is there a purpose for 

``` 
let currTrip = this.props.currTripInfo.trip;
if (currTrip === null) {
      return;
}
```

it appears as though it is never being used?

Reply:
In order to access the trip_start variable, we need the trip to not be null.